pkg/sbom/cyclonedx/unmarshal.go

```diff
@@ -408,10 +408,22 @@ func toTrivyCdxComponent(component cdx.Component) ftypes.Component {
 	}
 }
 
-func getPackageName(typ string, component cdx.Component) string {
-	// Jar uses `Group` field for `GroupID`
-	if typ == packageurl.TypeMaven && component.Group != "" {
-		return fmt.Sprintf("%s:%s", component.Group, component.Name)
+func getPackageName(purl *purl.PackageURL, component cdx.Component) string {
+	if purl.Type == packageurl.TypeMaven {
+		return convertMavenPackage(purl, component)
 	}
 	return component.Name
```

Comment:
i rechecked code before this line and found that we don't parse CycloneDX without `purl` - https://github.com/aquasecurity/trivy/blob/eea3320d83c240f7e7bf9e4cece4197e39b868ff/pkg/sbom/cyclonedx/unmarshal.go#L337-L340

So we can remove this check (sorry for confusing you).

I think we can remove `purl == ""` check and move logic to  `getPackageName` and use [parsed package name](https://github.com/aquasecurity/trivy/blob/eea3320d83c240f7e7bf9e4cece4197e39b868ff/pkg/purl/purl.go#L84-L90) ([pkg.Name = getPackageName(p.Type, pkg.Name, component)](https://github.com/aquasecurity/trivy/blob/eea3320d83c240f7e7bf9e4cece4197e39b868ff/pkg/sbom/cyclonedx/unmarshal.go#L349)). Something like that:

```suggestion
func getPackageName(typ, pkgNameFromPurl string, component cdx.Component) string {
	if typ == packageurl.TypeMaven {
	        // Jar uses `Group` field for `GroupID`
		if component.Group != "" {
			return fmt.Sprintf("%s:%s", component.Group, component.Name)
		} else {
		        // use name derived from purl if `Group` doesn't exist
			return pkgNameFromPurl
		}
	}
	return component.Name
}
```

Reply:
Wow :) Great idea! Thanks a lot! Done :)